app/models/locations/map.rb

```diff
@@ -0,0 +1,56 @@
+module Locations
+  class Map
+    include ActiveModel::Model
+
+    def initialize(params = {})
+      @@locator_url = "http://library.princeton.edu/locator/index.php?"
+      @@stackmap_url = "http://princeton.stackmap.com/view/?"
+      @params = params
+    end
+
+    def url
+      if @params[:loc] && @params[:id]
+        holding_location = get_holding_location(@params[:loc])
+        bibrec = get_bibrec(@params[:id])
+        if !holding_location.nil? && !bibrec.nil?
+          lib = get_lib(holding_location.locations_library_id)
+          if lib.code == 'firestone'
+            @@locator_url + "loc=" + @params[:loc]+ "&id=" + @params[:id]
+          else
```

Comment:
Yes, in theory all the locations in off-site facilities and closed stack locations should have that flag return false.